src/main/java/frc/robot/Robot.java

```diff
@@ -165,11 +143,12 @@ public void autonomousInit() {
     // Set motors to brake for the drive train
     m_robotContainer.robotDrive.setIdleMode(IdleMode.kBrake);
 
-    m_autoCommandInterface = chosenAuto.getSelected();
     //m_robotContainer.carouselCommand.schedule();
-    // schedule the autonomous command (example)
-    if (m_autoCommandInterface != null) {
-      m_autoCommandInterface.schedule();
+
+    // schedule the autonomous command
+    m_autonomousCommand = chosenAuto.getSelected();
```

Comment:
I'm not sure I like the implicit type conversion here. Even though it's legal, it could be confusing.

Reply:
In autoInit and teleopInit, the only relevant feature is that it is Command, so it does not seem necessary to save it as a AutoCommand. Yes, a little loss of info, but in some ways safer.

But whatever, I will change that class.

Reply:
"m_autonomousCommand" is now type AutoCommandInterface, so no conversion.